src/datasets/utils/py_utils.py

```diff
@@ -188,7 +187,19 @@ def _asdict_inner(obj):
     return _asdict_inner(obj)
 
 
-@contextlib.contextmanager
+@contextmanager
+def nullcontext():
+    """Context manager that does no additional processing.
+    Used as a stand-in for a normal context manager, when a particular
+    block of code is only sometimes used with a normal context manager:
+    cm = optional_cm if condition else nullcontext()
+    with cm:
+        # Perform operation, using optional_cm if condition is True
+    """
+    yield
```

Comment:
3.7 is a minimal supported Python version as of yesterday :), so this can be replaced with `contextlib.nullcontext`

Reply:
Nice, just did the change :)